kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyExceptionMapper.java

```diff
@@ -197,57 +189,15 @@
  */
 public class KafkaProxyExceptionMapper {
 
-    private final ConcurrentMap<Class<? extends Throwable>, Function<? extends Throwable, Optional<ResponseFrame>>> responsesByExceptionType;
-
-    public KafkaProxyExceptionMapper() {
-        responsesByExceptionType = new ConcurrentHashMap<>();
-    }
-
-    /**
-     * Register a function to be invoked when the particular throwable is encountered.
-     * @param throwableClass the target class to register
-     * @param responseFunction the function to invoke when the throwable is encountered
-     */
-    public <E extends Throwable> void registerExceptionResponse(Class<E> throwableClass,
-                                                                Function<E, Optional<ResponseFrame>> responseFunction) {
-        responsesByExceptionType.put(throwableClass, responseFunction);
-    }
-
-    /**
-     * Maps a given throwable by applying the function registered for the shallowest cause.
-     * @param throwable to have its cause chain searched.
-     * @return empty if the connection should be closed without any response,
-     * otherwise a non-empty (and not null) response to send to the client.
-     */
-    @SuppressWarnings("java:S1452")
-    public Optional<ResponseFrame> mapException(Throwable throwable) {
-        var candidate = throwable;
-        Set<Throwable> visitedExceptions = Collections.newSetFromMap(new IdentityHashMap<>());
-        while (candidate != null
-                && visitedExceptions.add(candidate)) { // Exit the loop when we see an exception we saw already
-
-            final var mappedFunction = containsMapping(candidate);
-            if (mappedFunction.isPresent()) {
-                return mappedFunction.get().apply(candidate);
-            }
-            candidate = candidate.getCause();
-        }
-        return Optional.empty();
-    }
-
-    private <E extends Throwable> Optional<? extends Function<? super E, Optional<ResponseFrame>>> containsMapping(E localCause) {
-        return responsesByExceptionType.entrySet().stream()
-                .filter(entry -> entry.getKey().isInstance(localCause))
-                .map(entry -> (Function<? super E, Optional<ResponseFrame>>) entry.getValue())
-                .findFirst();
+    private KafkaProxyExceptionMapper() {
```

Comment:
As a matter of style I don't like static util classes which is what this becomes in this PR. 

So while I accept there are limitations to the mapping logic this PR removes (for example it doesn't select the most specific mapper for each exception class), I don't think debatable improvements in readability of the exception handling out weigh the increased pain in testability.